acbuild/isolator.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2015 The appc Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/appc/acbuild/Godeps/_workspace/src/github.com/spf13/cobra"
+)
+
+var (
+	cmdIso = &cobra.Command{
+		Use:   "isolator [command]",
+		Short: "Manage isolators",
+	}
+	cmdAddIso = &cobra.Command{
+		Use:     "add NAME JSON_FILE",
+		Short:   "Add an isolator",
+		Long:    "Updates the ACI to contain an isolator with the given name and value. If the isolator exists, its value will be changed.",
+		Example: "acbuild isolator add resource/cpu ./value.json",
+		Run:     runWrapper(runAddIso),
+	}
+	cmdRmIso = &cobra.Command{
+		Use:     "remove NAME",
+		Aliases: []string{"rm"},
+		Short:   "Remove an isolator",
+		Long:    "Updates the current ACI's manifest to not contain the given isolator",
+		Example: "acbuild isolator remove resource/memory",
+		Run:     runWrapper(runRemoveIso),
+	}
+)
+
+func init() {
+	cmdAcbuild.AddCommand(cmdIso)
+	cmdIso.AddCommand(cmdAddIso)
+	cmdIso.AddCommand(cmdRmIso)
+}
+
+func runAddIso(cmd *cobra.Command, args []string) (exit int) {
+	if len(args) == 0 {
+		cmd.Usage()
+		return 1
+	}
+	if len(args) != 2 {
+		stderr("isolator add: incorrect number of arguments")
+		return 1
+	}
+
+	var val []byte
+	var err error
+	if args[1] == "-" {
```

Comment:
This is kind of a nit, but my preference for structuring this sort of thing is to switch on the interface rather than all the logic
```
var r io.Reader
var err error
if args[1] == "-" {
  r = os.Stdin
} else {
  r, err = os.Open(args[1])
  ...
}
```